bugtool/configuration/configuration.go

```diff
@@ -0,0 +1,379 @@
+	sort.Strings(arr)
+	if len(arr) <= 1 {
+		return arr
+	}
+	narr := []string{arr[0]}
+	for i := 1; i < len(arr); i++ {
+		if arr[i] != arr[i-1] {
+			narr = append(narr, arr[i])
+		}
+	}
+	return narr
+}
+
+// CreateDump creates the root dump task to be used for bugtool dumps.
+// If a configuration file is specified, it will attempt to use that
+// to load a task, otherwise it uses the "topics" list to compose a
+// a task.
+func CreateDump(conf *options.Config) dump.Task {
+	if conf.ConfigFile != "" && !(conf.Generate || conf.DryRun) {
+		log.Infof("Using specified config file: %s", conf.ConfigFile)
+		fd, err := os.Open(conf.ConfigFile)
+		if err != nil {
+			log.Fatalf("Failed to open config file %s: %v", conf.ConfigFile, err)
+		}
+		t, err := dump.Decode(fd)
+		if err != nil {
+			log.Fatalf("Failed to decode config %s: %v", conf.ConfigFile, err)
+		}
+		return t
+	}
+
+	// Handle deprecated topic specific flags.
+	if conf.EnvoyDump {
+		conf.Topics = append(conf.Topics, "envoy")
+	}
+
+	if conf.GetPProf {
+		log.Info("[Deprecated] Use --topics=pprof instead")
+		conf.Topics = append(conf.Topics, "pprof")
+	}
+
+	ts := dump.Tasks{}
+	conf.Topics = deduplicate(conf.Topics)
+	log.Debugf("Final topics list %v", conf.Topics)
+	for _, topic := range conf.Topics {
+		switch topic {
+		// Bpfmaps contains dump tasks for dumping Cilium bpf maps,
+		// as well as a list of bpf maps/attachments/programs.
+		//
+		// Anything dumped with 'bpftool' should go in here.
+		case "bpfmaps":
+			// bpfmaps is a dump of Cilium related bpf maps.
+			ts = append(ts,
+				dump.NewDir("bpfmaps", GenerateBPFToolTasks()),
+			)
+
+		// Agent contains output from cilium agent commands (i.e. cilium agent API).
+		// It also includes state data.
+		//
+		// Anything dumped with the Pods "cilium ..." command belongs in here.
+		case "agent":
+			// agent is dumps of data gathered directly from the
+			// Cilium Agent.
+			ts = append(ts,
+				dump.NewDir("agent", CiliumTasks()),
+			)
+			fallthrough
+
+		// Runtime state contains cilium runtime state that is written to disk,
+		// including endpoint restore data.
+		// This is also included as part of the "agent" topic.
+		case "state":
+			// Append with . will include only contents of directory.
+			runFiles := dump.NewFile("/run/cilium/state/.")
+			if conf.ExcludeObjectFiles {
+				runFiles = runFiles.WithExclude("/[0-9]*/*.o")
+			}
+			ts = append(ts, dump.NewDir("state", dump.Tasks{runFiles}))
+
+		// System includes output to host system commands and configurations.
+		// Command output is organized by structured/unstructured for commands
+		// that has a easily parsable output (i.e. JSON, YAML) and anything else.
+		case "system":
+			logs, unstructured, structured := defaultResources()
+			// system contains system level dumps.
+			ts = append(ts, dump.NewDir("system", dump.Tasks{
+				dump.NewDir("structured", structured),
+				dump.NewDir("unstructured", unstructured),
```

Comment:
Why separate structured and unstructured command outputs? As a user, I typically go into the bugtool report looking for some specific information. I have an idea of which command will give me that information, but I won't remember if its output is among the structured or unstructured ones :-/

Reply:
Yeah I think you're right, I combined them into a single directory: `cmd`.